src/util/memory.h

```diff
@@ -52,13 +52,13 @@
    #elif defined(__GNUC__)
       // std::make_unique was added in gcc 4.9
       #define GCC_VERSION (__GNUC__ * 10000 + __GNUC_MINOR__ * 100 + __GNUC_PATCHLEVEL__)
-      #if GCC_VERSION >= 40900
+      #if GCC_VERSION >= 40900 && __cplusplus > 201103L
```

Comment:
Indeed, compiling with -std=c++14 should pull in <memory> from GCC instead 
of this workaround. But I have not tested it.

I've also added an additional commit with a comment why we need to check 
__cplusplus here.

On 08/28/2015 12:39 AM, RJ Ryan wrote:
>
> In src/util/memory.h 
> <https://github.com/mixxxdj/mixxx/pull/693#discussion_r38154927>:
>
> > @@ -52,13 +52,13 @@
> >     #elif defined(__GNUC__)
> >        // std::make_unique was added in gcc 4.9
> >        #define GCC_VERSION (__GNUC__ * 10000 + __GNUC_MINOR__ * 100 + __GNUC_PATCHLEVEL__)
> > -      #if GCC_VERSION >= 40900
> > +      #if GCC_VERSION >= 40900 && __cplusplus > 201103L
>
> Sorry, missed the mention in the PR description. So this would prevent 
> defining that the compiler supports make_unique when the user specified 
> -std=c++04 for example?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/693/files#r38154927>.
>